Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp

```diff
@@ -73,28 +82,44 @@ SpectralFieldData::SpectralFieldData( const amrex::BoxArray& realspace_ba,
         cufftResult result;
 #if (AMREX_SPACEDIM == 3)
         result = cufftPlan3d( &forward_plan[mfi], fft_size[2],
+#    ifdef AMREX_USE_FLOAT
+                              fft_size[1],fft_size[0], CUFFT_R2C);
+#    else
                               fft_size[1],fft_size[0], CUFFT_D2Z);
+#    endif
```

Comment:
I think this can be
`result = cufftPlan3d( &forward_plan[mfi], fft_size[2], fft_size[1], fft_size[0],
#ifdef AMREX_USE_FLOAT
                                  CUFFT_R2C);
#else 
                                  CUFFT_D2Z);
#endif `

Reply:
Also, @MaxThevenet , thanks for teaching me how to write this correctly in the review :)